### ISSUE TITLE
Replace featured image text isn't visible.

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66005]
### Description

See related https://github.com/WordPress/gutenberg/issues/62743 and https://github.com/WordPress/gutenberg/pull/65747

Reported on Trac by [Necarie](https://profiles.wordpress.org/necarie/).

I'm using a fresh install of WordPress with no additional themes or plugins installed. I'm using the Twenty Twenty-Four theme and the latest version of Firefox.

I imported an xml file to restore posts from another WordPress. There were media files referenced in it that failed to upload, which was fine as I intended to replace the images manually. When I edited the posts to replace the featured images, the featured images option seemed to be missing.

I eventually figured out the option was there, but I couldn't see it because the text on the buttons was the same color as the background. This only happens when there is a reference to a missing image - once I added a valid image, the button appeared as black text on a light background.

It may not seem like a big deal, but it took me a long time to figure out because I'm not familiar with this version of WordPress or where the button was supposed to be. I hope that by reporting it I can spare someone else the frustration I felt - it seems like something easy to fix.

### Step-by-step reproduction instructions

- Import an XML file with posts that have featured images set.
- Do not copy over the actual images to the `wp-content/uploads` directory so that the images are actually missing but the posts meta data still refers to them.
- Edit a post with a featured image set.
- Observe the Featured Image UI in the Post summary panel.

### Screenshots, screen recording, code snippet

This shows the button where "Replace Image" should be visible, but it isn't.

![Image](https://github.com/user-attachments/assets/cd8ef611-518f-46e3-83f1-642917cb2f9a)

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by afercia** [comment]
I can reproduce the same issue in a slightly different case e.g. deleting from the filesystem an image that was already set as featured image. The UI behaves slightly differently but it is still very unclear as it doesn't take into account missing or broken images.

First of all I'd like to point out that browsers behave differently when the path to an image is broken. The different behavior depends on various factors e.g. the presence of an alt text, the size applied to the `<img />` element and such.

Basically, when there's no alt text, Chrome tends to display a sort of 'broken image placeholder', while Firefox doesn't. Screenshot: 
- On the left: Chrome
- On the right: Firefox

Worth noting [Necarie](https://profiles.wordpress.org/necarie/) reported to use Firefox.

![Image](https://github.com/user-attachments/assets/1dbb0f99-2616-4b75-8e97-3a856e33921f)

In my test the UI behavior is slightly different because the image placeholder is bigger. I guess because my image was actually already uploaded to the Media Library and I deleted it _after_ it was set. Instead, in the reporter case, the image placeholder is way smaller and appears to be in the 'loading' stage.

Regardless, in both cases the placeholder doesn't show anything:
- The featured image alt is forced [to and empty string](https://github.com/WordPress/gutenberg/blob/03cfbff40ef2e5bf9ad369bb0a18642aa70833ef/packages/editor/src/components/post-featured-image/index.js#L191). This should be reconsidered, as one of the purposes of the alt text is, well, to allow browsers to show the alt text when the image is missing. That's why it is called alternative text.
- The Replace and Remove buttons appear only on hover or focus, which is extremely confusing when the image is missing. See related https://github.com/WordPress/gutenberg/issues/62743 and https://github.com/WordPress/gutenberg/pull/65747

It's important to note that a missing image was problematic also with the previous featured image UI in WordPress 6.5. Screenshot:

![Image](https://github.com/user-attachments/assets/d4726d6c-5af1-4fc2-96c0-87fab2102529)

Only the UI in WordPress 6.3 was still clearly operable with a missing image. The Replace and Remove controls where always visible, thus making the UI clearer and always operable. Screenshot:

![Image](https://github.com/user-attachments/assets/09a7aa78-b684-4f96-9a06-9f37fdb8a716)

## Proposal
While the featured image UI provides meaningful labeling and description of the image on the _placeholder button_, the forced empty alt text doesn't seem ideal. I'd like to propose to use the same description of the button for the image alt text so that when an image is missing the alt text would work as intended. Screenshot:

![Image](https://github.com/user-attachments/assets/c10ec7e3-a016-4268-9a6f-b3c06548c08f)

On top of that, I think the Replace and Remove buttons should always be visible as proposed in https://github.com/WordPress/gutenberg/issues/62743 and https://github.com/WordPress/gutenberg/pull/65747

**Comment by afercia** [comment]
Part of this issue has been addressed in https://github.com/WordPress/gutenberg/pull/66189

**Comment by afercia** [comment]
I was able to reproduce the originally reported issue by importing an xml file with a post containing a featured image and then skipping the 'Download and import file attachments`. Basically, the UI doesn't take into account the case when the attachment ID is saved in the database but there's no `media` object containing the image meta data. 

In this case, the UI does render the placeholder button but it does not render the image element. The placeholder button is empty:

https://github.com/WordPress/gutenberg/blob/f3e8f4a5c11a8bf2f7b4e8e2ff3c7963c23194f3/packages/editor/src/components/post-featured-image/index.js#L202-L208

As such, there is no `alt` attribute to use to provide some message. The button itself should contain some meaningful message.

Animated GIF to illustrate the UI state when this occurs:


![Image](https://github.com/user-attachments/assets/df2b6f88-0732-4ca2-a02a-d23baa647def)
